Fix getWellKnownSolid with e2e tests and custom jsonld loader

ThisIsMissEm opened the pull request:
This is just to see if it's working correctly or not across different environments, as I suspect we're stubbing too much in the unit tests.

ThisIsMissEm committed:
Add e2e test for getWellKnownSolid

ThisIsMissEm committed:
chore: fix warning in e2e tests for already loaded environment

matthieubosquet reviewed (approved):
I checked the implementation of jsonld.

This looks good.

matthieubosquet commented on src/formats/jsonLd.test.ts:
```diff
@@ -144,4 +150,94 @@ describe("The Parser", () => {
+          [RequestInfo, RequestInit?]
+        >;
+      };
+      mockedFetcher.fetch.mockResolvedValueOnce(
+        new Response(
+          JSON.stringify({
+            "@context": {
+              pim: "http://www.w3.org/ns/pim/space#",
+              "@version": 1.1,
+              "@protected": true,
+              id: "@id",
+              type: "@type",
+              storage: { "@id": "pim:storage", "@type": "@id" },
+            },
+          }),
+          { headers: { "Content-Type": "application/json" } }
+        )
+      );
+
+      const jsonLd = `
+      {
+        "@context":"https://pod.inrupt.com/solid/v1",
+        "storage":"https://pod.inrupt.com/username/"
+      }
+    `;
+
+      await parser.parse(jsonLd, {
+        internal_resourceInfo: {
+          sourceIri: "https://some.pod/resource",
+          isRawData: false,
+          linkedResources: {},
+        },
+      });
```
Nice :)

matthieubosquet reviewed (approved):
It looks good.

I was not familiar with the jsonld module which I checked, well done figuring out how to overload the document loader!

ThisIsMissEm committed:
fix: getWellKnownSolid by using custom jsonld loader

This bypasses the @digitalbazaar/http-client package using our own fetch implementation instead
